aspen/http/resource.py

```diff
@@ -7,25 +8,18 @@ class Static(object):
     """
 
     def __init__(self, request_processor, fspath, raw, media_type):
-        self.request_processor = request_processor
+        assert type(raw) is bytes  # sanity check
         self.raw = raw
         self.media_type = media_type
         if media_type == 'application/json':
-            self.media_type = self.request_processor.media_type_json
+            self.media_type = request_processor.media_type_json
+        if self.media_type.startswith('text/'):  # XXX this is bad
```

Comment:
Or, we could make `charset` a property of the `resource.{Static,Dynamic}` classes instead of a global configuration value. That probably fits better with our evolution away from an app server towards being just a library.

```python
USE_DEFAULT_CHARSET = object()

class Static(object):
    charset = None
    def __init__(self, ..., charset=USE_CLASS_DEFAULT_CHARSET):
        pass

class Dynamic(Simplate):
    charset = 'UTF-8'
    def __init__(self, ..., charset=USE_CLASS_DEFAULT_CHARSET):
        pass
```